### PR TITLE
brunorosano/feature/BE-001

### PR DESCRIPTION
**Pull request by brunorosano**
1.  Create patient and admin entities.
2. Draft doctor and specialty entities.
3. Create admin CRUD.

### CODE REVIEW

File: src/main/java/br/unicamp/cecom/appointmentscheduler/core/exception/NotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package br.unicamp.cecom.appointmentscheduler.core.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(value = HttpStatus.NOT_FOUND)
+public class NotFoundException extends RuntimeException {
+
+    private static final long serialVersionUID = -4438072051802392231L;
```
Comment (caiopardal) [inline src/main/java/br/unicamp/cecom/appointmentscheduler/core/exception/NotFoundException.java:9]: doubt: This is a leftover right? If so, please remove it :)

File: src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminRepository.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+public class AdminRepository {
+
+    private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+    private final JdbcTemplate jdbcTemplate;
+    private static final AdminRowMapper ROW_MAPPER = new AdminRowMapper();
+
+    private static final String CREATE = "INSERT INTO admin " +
+            "(fullname, email, phone) VALUES (:fullname,:email,:phone) RETURNING admin_id ";
+    private static final String UPDATE = "UPDATE admin SET " +
+            "(fullname, email, phone) VALUES (:fullname,:email,:phone) WHERE id =:adminId ";
```
Comment (caiopardal) [inline src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminRepository.java]: :adminId or :admin_id? Here we have both of them :/

File: src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminRepository.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    private static final String DELETE = "DELETE FROM admin WHERE admin_id = :adminId ";
+    private static final String FIND_BY_ID = "SELECT * FROM admin WHERE admin_id = ? ";
```
Comment (caiopardal) [inline src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminRepository.java]: missing argument at the "WHERE" statement?

File: src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminService.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    private final AdminRepository adminRepository;
+
+    public UUID create(final CreateAdminRequest request){
+        return adminRepository.create(request);
+    }
+
+    public void update(final UUID adminId, final UpdateAdminRequest request) {
+        if (this.adminRepository.update(adminId, request) == 0) {
+            throw new NotFoundException("message.admin.notfound");
```
Comment (caiopardal) [inline src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminService.java]:
```suggestion
            throw new NotFoundException("message.admin.notFound");
```

File: src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminService.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        }
+    }
+
+    public void delete(final UUID adminId) {
+        if (this.adminRepository.delete(adminId) == 0) {
+            throw new NotFoundException("message.admin.notfound");
```
Comment (caiopardal) [inline src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminService.java]:
```suggestion
            throw new NotFoundException("message.admin.notFound");
```

File: src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminService.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        }
+    }
+
+    public Optional<AdminEntity> findById(final UUID adminId) {
```
Comment (caiopardal) [inline src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminService.java:56]: Maybe we should add the not found exception here as well

File: src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/specialty/SpecialtyEntity.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+public class SpecialtyEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer specialtyId;
+
+    @NotBlank(message = "Specialty name must not be null and must contain at least one non-whitespace character")
+    @Max(value = 255, message = "Specialty name must have a maximum of 255 characters")
+    private String specialtyName;
+}
```
Comment (caiopardal) [inline src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/specialty/SpecialtyEntity.java]: nit: missing final line

**Review by caiopardal — CHANGES_REQUESTED** [review]
Just some syntax changes

File: src/main/java/br/unicamp/cecom/appointmentscheduler/core/exception/NotFoundException.java (hunk 1 above)
Comment (caiopardal) [reply src/main/java/br/unicamp/cecom/appointmentscheduler/core/exception/NotFoundException.java:9]: https://pt.stackoverflow.com/questions/96903/o-que-significa-serialversionuid

Comment (gabrieloswaldo) [reply src/main/java/br/unicamp/cecom/appointmentscheduler/core/exception/NotFoundException.java:9]: reference: https://pt.stackoverflow.com/questions/96903/o-que-significa-serialversionuid

File: src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminRestController.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.Optional;
+import java.util.UUID;
+
+import static java.lang.String.format;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.springframework.http.ResponseEntity.created;
+import static org.springframework.http.ResponseEntity.noContent;
+
+@RestController
+@RequiredArgsConstructor
```
Comment (gabrieloswaldo) [inline src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminRestController.java]: Add autowired for dependency injection:
```suggestion
@RequiredArgsConstructor(onConstructor = @__(@Autowired))
```

File: src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminService.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
```
Comment (gabrieloswaldo) [inline src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminService.java]: Add autowired for dependency injection:
```suggestion
@RequiredArgsConstructor(onConstructor = @__(@Autowired))
```

File: src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminService.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    public Optional<AdminEntity> findById(final UUID adminId) {
+        try {
+            return adminRepository.findById(adminId);
+        } catch (EntityNotFoundException e) {
+            throw new NotFoundException("message.admin.notFound");
+        }
+
+    }
```
Comment (gabrieloswaldo) [inline src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/admin/AdminService.java:62]: Skip line here :)

File: src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/doctor/DoctorEntity.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+import javax.validation.constraints.Max;
+import javax.validation.constraints.NotBlank;
+import java.util.UUID;
+
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class DoctorEntity {
```
Comment (gabrieloswaldo) [inline src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/doctor/DoctorEntity.java:19]: Add the annotation `@Table(name = "doctor")`

File: src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/patient/PatientEntity.java [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+import javax.validation.constraints.Max;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Size;
+
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class PatientEntity {
```
Comment (gabrieloswaldo) [inline src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/patient/PatientEntity.java:21]: Add the annotation `@Table(name = "patient")`

File: src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/specialty/SpecialtyEntity.java [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.NotBlank;
+
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
```
Comment (gabrieloswaldo) [inline src/main/java/br/unicamp/cecom/appointmentscheduler/core/features/specialty/SpecialtyEntity.java:20]: Add the annotation `@Table(name = "specialty")`

**Review by gabrieloswaldo — CHANGES_REQUESTED** [review]

**Pull request merged**